### DISCUSSION TITLE
Want to contribute a template to the official template gallery?

### DISCUSSION THREAD

**Discussion by martin-hasura** [hasura/template-gallery#2] [Announcements]
We'll be publishing a spec very soon for creating your own templates for the sample repository.

For the time being feel free to leave your ideas for template gallery projects in this discussion.

-----

PRs to the gallery are welcome - feel free to copy off one of our existing templates to create your own template. One of our maintainers will be happy to take a look, validate the template, and help prepare it for production.

The metadata is currently generated mainly by using the `Export Metadata` from the Hasura dashboard, and migrations are created using raw SQL.

**Comment by futuregridtechnologies** [comment]
1. Creating a SQL and graphql based rule engine that can compute nested and/or conditions and is compatible with subscriptions.
2. Boilerplate for Hasura actions that allows to send emails through hasura graphql, same for sms

**Reply by martin-hasura** [reply]
Hey there - for the first - just was wondering what kind of and/or conditions you'd be looking for which aren't currently support using the GraphQL layer?

For example - you could have a nested and/or condition using something like (using the Permissions w/ Attributes template):
```gql
subscription returnChat{
  _attributes_chat (
    where: {
      _and: [
        {
          message: {_is_null: false}
        },
        {
          _or: [
      	    { _and: { receiver_id: {_eq: 6}, created_at: {_gte: "2020-01-01 01:01:01.000000"} } }
      	    { _and: { sender_id: {_eq: 6}, created_at: {_gte: "2020-01-01 01:01:01.000000"} } }
    	  ]
        },
      ]
    }  
  ) {
    id
    message
    sender_id
    receiver_id
    created_at
  }
}
```
For the 2nd - Actions and Remote Schemas are on the roadmap for support for the template gallery. We're currently looking at adding those with parameterized templates so that when you add an action you can specify where to point it at in terms of endpoint.

If you're interested in an email Action Praveen has a really good example you can have a look at over here as well:
https://www.youtube.com/watch?v=iVoS6Ti2QHQ
https://github.com/hasura/hasura-actions-examples/tree/master/email
